### PR TITLE
MRC-1208 custom query parameters

### PR DESCRIPTION
**Pull request by bwell-bruce**

### CODE REVIEW

File: src/dist/js/search.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 const formElement = document.getElementById('searchForm');
 formElement.addEventListener('submit', searchSubmit);
+
+const advSearchFormElement = document.getElementById('advSearchForm');
+advSearchFormElement.addEventListener('submit', searchSubmit);
+
 const resetButtonElement = document
```
Comment (DarylCheung) [inline src/dist/js/search.js:7]: Lets update this one here too to be consistent with your current format

File: src/dist/css/app.css [hunk 2]
```diff
@@ -114,10 +114,34 @@ a.fa {
...
+  background-image: url(../../images/helix.png);
+  background-position: center center;
+  background-repeat: no-repeat;
+  background-size: contain;
+  display: inline-block;
+  height: 35px;
+}
+
+#advSearchButton {
+  position: absolute;
```
Comment (DarylCheung) [inline src/dist/css/app.css:140]: does having absolute here mess with the sizing of the page?

Comment (bwell-bruce) [reply src/dist/css/app.css:140]: @DarylCheung No, it stays contained/relative within the form element

File: src/utils/searchForm.util.js [hunk 3]
```diff
@@ -79,6 +80,31 @@ const getFormData = (req, resourceName) => {
...
+  const searchParams = getSearchParams(req);
+  const params = searchParams ? Object.fromEntries(searchParams.entries()) : {};
+  let advFormData = [];
+  const resourceFields = advSearchJson.entry.filter((entry) => {
+    return entry.resource.base.includes(resourceName) && entry.resource.type === 'string';
+  });
+
+  resourceFields.forEach((advParam) => {
+    const foundBasic = basicFormData.find((formData) => formData.name === advParam.resource.name);
+    if (!foundBasic) {
```
Comment (DarylCheung) [inline src/utils/searchForm.util.js:94]: Could we invert the if here to be if (foundBasic) return;, to reduce nesting

File: src/dist/js/search.js (hunk 1 above)
Comment (bwell-bruce) [reply src/dist/js/search.js]: @DarylCheung This is updated

File: src/utils/searchForm.util.js (hunk 3 above)
Comment (bwell-bruce) [reply src/utils/searchForm.util.js]: Updated

**Review by DarylCheung — APPROVED** [review]

**Pull request merged**